lib/hl7/message.ex

```diff
@@ -55,9 +54,10 @@ defmodule HL7.Message do
     [encoding_characters | _] = msh_tail
     msh_without_field_separator = [name | msh_tail]
 
-    [component, repeat, _escape_char, subcomponent] = String.graphemes(encoding_characters)
+    [component, repeat, _escape_char, subcomponent | truncation_char] =
+      String.graphemes(encoding_characters)
 
-    join_by_character_list = [field_separator, repeat, component, subcomponent]
+    join_by_character_list = [field_separator, repeat, component, subcomponent, truncation_char]
```

Comment:
I read the spec on this -- and I don't believe the truncation character should be used for splitting/joining at all. It just informs end systems that the data has been truncated at the last character (for systems that can't consume larger payloads). I believe the only change needed for this PR is to accept the header data with or without the truncator and rebuild it as the string.

Reply:
I don't code is doing any split on truncation character. Since truncation character is part of `encoding_characters` now, its just taking that into account and rebuild the same message. I will double check on it

Reply:
I see what you are saying, I'll remove this from this list